tdms/tests/system/data/input_generation/generate_test_input.py

```diff
@@ -0,0 +1,242 @@
+import os
+from glob import glob
+from pathlib import Path
+from subprocess import PIPE, Popen
+from typing import Union
+
+import yaml
+
+LOCATION_OF_THIS_FILE = os.path.dirname(os.path.abspath(__file__))
+# MATLAB executable path
+MATLAB_PATH = "matlab"
+# Additional options for running matlab on the command-line
+MATLAB_OPTS = ["-nodisplay", "-nodesktop", "-nosplash", "-r"]
+# Paths to matlab functions not in LOCATION_OF_THIS_FILE
+MATLAB_EXTRA_PATHS = [
+    os.path.abspath(LOCATION_OF_THIS_FILE + "/bscan"),
+    os.path.abspath(LOCATION_OF_THIS_FILE + "/matlab"),
+]
+
+
+class BScanArguments:
```

Comment:
This, for example, might be better as a simple function returning a dictionary.

Reply:
If I can get [`matlabengine`](https://uk.mathworks.com/help/matlab/matlab_external/install-the-matlab-engine-for-python.html) working then 100% yes. 

But otherwise, I wanted it to be it's own class just so that the messy "combine the fields to make the bscan argument call string" is all in one place, and I don't have to do it somewhere else.